src/Libraries/Microsoft.Extensions.AI/ChatCompletion/ChatClientStructuredOutputExtensions.cs

```diff
@@ -176,4 +190,31 @@ public static async Task<ChatCompletion<T>> CompleteAsync<T>(
             }
         }
     }
+
+    private static bool SchemaRepresentsObject(JsonElement schemaElement)
+    {
+        if (schemaElement.ValueKind is JsonValueKind.Object)
+        {
+            foreach (var property in schemaElement.EnumerateObject())
+            {
+                if (property.NameEquals("type"u8))
+                {
+                    return property.Value.ValueKind == JsonValueKind.String
+                        && property.Value.ValueEquals("object"u8);
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private static JsonNode? JsonElementToJsonNode(JsonElement element)
+    {
+        return element.ValueKind switch
+        {
+            JsonValueKind.Array => JsonArray.Create(element),
```

Comment:
My initial workaround was missing handling for null. Not applicable to schemas, but adding for completeness in case it gets copied over to other contexts.
```suggestion
            JsonValueKind.Null => null,
            JsonValueKind.Array => JsonArray.Create(element),
```